iothub/service/src/Amqp/ClientWebSocketTransport.cs

```diff
@@ -224,9 +224,9 @@ private async Task<int> ReadImplAsync(TransportAsyncCallbackArgs args)
             {
                 throw new IOException(httpListenerException.Message, httpListenerException);
             }
-            catch (TaskCanceledException taskCanceledException)
+            catch (TaskCanceledException)
             {
-                throw new TimeoutException(taskCanceledException.Message, taskCanceledException);
+                throw;
```

Comment:
Why catch the exception just to throw it again?

Reply:
Ah good catch, I should simply stop catching it here.